src/Sphere.jl

```diff
@@ -156,3 +156,75 @@ function normal_tvector_distribution(S::Sphere, x, σ)
     d = Distributions.MvNormal(zero(x), σ)
     return ProjectedFVectorDistribution(TangentBundleFibers(S), x, d, project_vector!, x)
 end
+
+struct GeodesicInterpolationMethod <: AbstractMethod end
+
+"""
+    mean!(
+        S::Sphere,
+        y,
+        x::AbstractVector,
+        w::AbstractWeights,
+        method::GeodesicInterpolationMethod;
+        shuffle_rng=nothing,
+        kwargs...,
+    )
+
+Compute the Riemannian mean (center of mass) of `x`. The algorithm is adapted
+from the below paper to handle the weighted mean. It uses repeated weighted
+geodesic interpolation and converges to the Riemannian mean when all `x` are
+within a distance of π/2 of the mean.
+
+If `shuffle_rng` is provided, it is used to shuffle the order in which the
+points are considered for computing the mean.
+
+For more information on the geodesic interpolation method, see the below paper:
+> Salehian H et al: An efficient recursive estimator of the
```

Comment:
I would prefer to see all the authors. There is not really a need to shorten that (unless its more than 12).